### DISCUSSION TITLE
Interpretation of per-residue representations

### DISCUSSION THREAD

**Discussion by ziruiwang1996** [facebookresearch/esm#172] [Q&A]
Hi:

Congrats on this amazing work! 

I have a question about the token representation I got back using esm_msa1b_t12_100M_UR50S(). I tested this model with 2 sequences with a length of 10 amino acids. I expected to get a token representation with shape 1 tensor x 2 seqs x 10 amino acids x 768 embedding dim. However, I got a tensor back with a shape 1 x 2 x 11 x 768. I wonder what is that additional vector in token representations comes from. 

Thanks!

**Comment by tomsercu** [comment]
Depending on the precise code you execute, I'd guess this is due to prepending bos in BatchConverter, which is matching how the model was trained. happens here: https://github.com/facebookresearch/esm/blob/main/esm/data.py#L276

**Reply by ziruiwang1996** [reply]
Thank you! I am going to study the code you mentioned to find out why.
